### PR TITLE
Add `path` and `readonly` options to `cupyx.optimizing.optimize`

### PR DESCRIPTION
**Pull request by kmaehashi**
Implements #3574.

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: cupyx/optimizing/_optimize.py [hunk 1]
```diff
@@ -83,7 +90,18 @@ def optimize(*, key=None, **config_dict):
...
     _optimize_config.set_current_context(context)
 
+    if path is not None:
+        if os.path.exists(path):
+            context.load(path)
+        elif readonly:
+            warnings.warn('''
+The specified path {} could not be found, and the readonly option is set.
+The optimization results will never be stored.
+                '''.format(path))
```
Comment (asi1024) [inline cupyx/optimizing/_optimize.py:100]: Remove trailing spaces from the warning message.
```suggestion
'''.format(path))
```

File: cupyx/optimizing/_optimize.py [hunk 2]
```diff
@@ -52,6 +54,11 @@ def optimize(*, key=None, **config_dict):
 
     Args:
         key (string or None): The cache key of optimizations.
+        path (string or None): The path to save optimization cache records.
+            When path is specified and exists, records will be loaded from
+            the path. When readonly option is set to False, optimization
+            cache records will be saved to the path after the optimization.
+        readonly (bool): See the description of path option.
```
Comment (asi1024) [inline cupyx/optimizing/_optimize.py:61]:
```suggestion
        readonly (bool): See the description of ``path`` option.
```

File: cupyx/optimizing/_optimize.py [hunk 3]
```diff
@@ -52,6 +54,11 @@ def optimize(*, key=None, **config_dict):
 
     Args:
         key (string or None): The cache key of optimizations.
+        path (string or None): The path to save optimization cache records.
+            When path is specified and exists, records will be loaded from
+            the path. When readonly option is set to False, optimization
```
Comment (asi1024) [inline cupyx/optimizing/_optimize.py:59]:
```suggestion
            the path. When readonly option is set to ``False``, optimization
```

**Comment by kmaehashi** [comment]
Thanks!
pfnCI, test this please.

**Review by asi1024 — APPROVED** [review]

**Pull request merged**